tests/shared/core/test_domain.py

```diff
@@ -368,6 +368,23 @@ def test_merge_yaml_domains():
     assert domain.session_config == SessionConfig(20, True)
 
 
+@pytest.mark.parametrize("default_intent", DEFAULT_INTENTS)
+def test_merge_yaml_domains_with_default_intents(default_intent: Text):
+    test_yaml_1 = """intents: []"""
+
+    # this domain contains an overridden default intent
+    test_yaml_2 = f"""intents:
+- greet
+- {default_intent}"""
+
+    domain_1 = Domain.from_yaml(test_yaml_1)
+    domain_2 = Domain.from_yaml(test_yaml_2)
+    domain = domain_1.merge(domain_2)
+
+    assert default_intent in domain.intents
```

Comment:
can we also test that the `default_intent` is dumped as part of `as_dict` ?

Reply:
👍